frontend/src/app/components/chat/messages/MessageWindow.jsx

```diff
@@ -34,7 +38,14 @@ const MessageWindow = () => {
         schema: 'public',
         table: 'messages',
       }, payload => {
-        setMessages(messages => [...messages, payload.new]);
+        setMessages(prevMessages => {
```

Comment:
set karna hi kyu hai? Isn't it better to check outside and then apply directly if state change is required